# Wildfire Size Class Prediction

In [15]:
import pandas as pd
import numpy as np
import pandas as pd
import datetime as dt
import sqlite3
import julian

## Load in data

In [10]:
# Select only pertinent features (this should make loading in data faster)
con = sqlite3.connect("wildfire_data.sqlite")
query="""
SELECT 
FIRE_NAME,
FIRE_SIZE_CLASS,
STAT_CAUSE_DESCR, STAT_CAUSE_CODE,
STATE, COUNTY,
LONGITUDE, LATITUDE,
DISCOVERY_DATE, 
DISCOVERY_TIME, 
CONT_DATE, 
CONT_TIME 
from Fires
"""
query=query.strip()
df = pd.read_sql_query(query, con)
con.close();

In [11]:
print(df.shape)
df.head()

(1880465, 12)


FIRE_NAME FIRE_SIZE_CLASS STAT_CAUSE_DESCR  STAT_CAUSE_CODE STATE COUNTY  \
0  FOUNTAIN               A    Miscellaneous              9.0    CA     63   
1    PIGEON               A        Lightning              1.0    CA     61   
2     SLACK               A   Debris Burning              5.0    CA     17   
3      DEER               A        Lightning              1.0    CA      3   
4  STEVENOT               A        Lightning              1.0    CA      3   

    LONGITUDE   LATITUDE  DISCOVERY_DATE DISCOVERY_TIME  CONT_DATE CONT_TIME  
0 -121.005833  40.036944       2453403.5           1300  2453403.5      1730  
1 -120.404444  38.933056       2453137.5           0845  2453137.5      1530  
2 -120.735556  38.984167       2453156.5           1921  2453156.5      2024  
3 -119.913333  38.559167       2453184.5           1600  2453189.5      1400  
4 -119.933056  38.559167       2453184.5           1600  2453189.5      1200

## Drop rows with missing data
Given our data is not sparse at all, we have the freedom to just drop all rows that are missing data we care about.

In [13]:
# drop the following rows if they have missing data for the following features
needed_cols = ['FIRE_SIZE_CLASS', 'DISCOVERY_DATE', 'DISCOVERY_TIME', 'CONT_DATE', 'CONT_TIME', 'STAT_CAUSE_CODE', 'STATE', 'LONGITUDE', 'LATITUDE']
df = df.dropna(subset=needed_cols) # remove rows where both of these are missing
df.shape

(892007, 12)

## Create new columns to work with

Map fire size class to integers so they can be ordered.

In [56]:
di = {"A": 1, "B": 2, "C": 3, "D": 4, "E": 5, "F": 6, "G": 7}
df['FIRE_SIZE_CLASS'] = df['FIRE_SIZE_CLASS'].map(di)

Map states to indices to help handle categorical

In [96]:
states = df.STATE.unique()
ind = states.argsort(axis=0)
state_di = {states[i]: i for i in ind}
df['STATE_CODE']=df['STATE'].map(state_di)

Convert the date/time columns to datetime objects. Originally they are in julian time. Also calculate the time to containment (time delta of containment date - discovery date)

In [16]:
#To make these dates and times easier to manage, let's convert them to datetime. We can add new columns DISCOVERY_DATETIME and CONTAINMENT_DATETIME.
df['DISCOVERY_DATETIME'] = df['DISCOVERY_DATE'];
df['CONT_DATETIME'] = df['CONT_DATE'];

In [17]:
#To populate those two rows, let's convert them into datetime.
df['DISCOVERY_DATETIME'] = df['DISCOVERY_DATETIME'].apply(lambda x: julian.from_jd(x, fmt="jd"))
df['CONT_DATETIME'] = df['CONT_DATETIME'].apply(lambda x:julian.from_jd(x, fmt="jd"))


In [18]:
#Let's also add the time 
temp_df = pd.DataFrame();
temp_df['dt'] = df['DISCOVERY_TIME'].apply(lambda x: dt.timedelta(hours=int(x[0:2]), minutes=int(x[2:5])))
df['DISCOVERY_DATETIME'] = df['DISCOVERY_DATETIME'] + temp_df['dt']
df['DISCOVERY_DATETIME'].head()

0   2005-02-02 13:00:00
1   2004-05-12 08:45:00
2   2004-05-31 19:21:00
3   2004-06-28 16:00:00
4   2004-06-28 16:00:00
Name: DISCOVERY_DATETIME, dtype: datetime64[ns]

In [19]:
#Do the same thing for CONT_DATETIME
temp_df = pd.DataFrame();
temp_df['dt'] = df['CONT_TIME'].apply(lambda x: dt.timedelta(hours=int(x[0:2]), minutes=int(x[2:5])))
df['CONT_DATETIME'] = df['CONT_DATETIME'] + temp_df['dt']
df['CONT_DATETIME'].head()

0   2005-02-02 17:30:00
1   2004-05-12 15:30:00
2   2004-05-31 20:24:00
3   2004-07-03 14:00:00
4   2004-07-03 12:00:00
Name: CONT_DATETIME, dtype: datetime64[ns]

In [20]:
df.to_csv("data_with_target.csv")
#Checkpoint to save the data with the above columns before reloading

In [5]:
df = pd.read_csv("data_with_target.csv")

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (9,11,12,13,14,15,16,17,18,19,36,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [21]:
dict(zip(df.STAT_CAUSE_CODE,df.STAT_CAUSE_DESCR))

{9.0: 'Miscellaneous',
 1.0: 'Lightning',
 5.0: 'Debris Burning',
 4.0: 'Campfire',
 2.0: 'Equipment Use',
 7.0: 'Arson',
 8.0: 'Children',
 6.0: 'Railroad',
 3.0: 'Smoking',
 11.0: 'Powerline',
 12.0: 'Structure',
 10.0: 'Fireworks',
 13.0: 'Missing/Undefined'}

In [25]:
df['TIME_TO_CONT'] = df['CONT_DATETIME'] - df['DISCOVERY_DATETIME']

df['TIME_TO_CONT'].describe()

count                     892007
mean      1 days 06:44:52.524004
std      13 days 19:33:02.720135
min              0 days 00:00:00
25%              0 days 00:30:00
50%              0 days 01:28:00
75%              0 days 04:45:00
max           3653 days 01:30:00
Name: TIME_TO_CONT, dtype: object

In [40]:
# convert to hours for easier sorting
df['HOURS_TO_CONT'] = df['TIME_TO_CONT'].apply(lambda x: round(float(x.total_seconds()/60/60)),2)
df.HOURS_TO_CONT.describe()

count    892007.000000
mean         30.728347
std         331.551520
min           0.000000
25%           0.000000
50%           1.000000
75%           5.000000
max       87674.000000
Name: HOURS_TO_CONT, dtype: float64

## Investigate the outliers in HOURS_TO_CONT 

75% is only 5 hours to cont, but max is abnormally high. 
Some dates seem to be incorrect (in record) -- month/day and time of day (h/m) are correct but wrong only in year.

Be careful because some differences in year are okay (Dec/Jan fires).

Since total dataset is 892007 instances, and of those, only 6000 are greater than 1000 hours to cont, these are definitely outliers and may need to be omitted from training for better accuracy/performance. We could also use a model that is insensitive to outliers.

In [71]:
df[df.HOURS_TO_CONT > 1000].HOURS_TO_CONT.describe()

count     5994.000000
mean      2289.123624
std       3257.553581
min       1001.000000
25%       1340.250000
50%       1776.000000
75%       2392.750000
max      87674.000000
Name: HOURS_TO_CONT, dtype: float64

In [61]:
df[(df.HOURS_TO_CONT > 8000) & (df.FIRE_SIZE_CLASS > 2)].sort_values(by='FIRE_SIZE_CLASS')

FIRE_NAME  FIRE_SIZE_CLASS   STAT_CAUSE_DESCR  STAT_CAUSE_CODE  \
182647               ML5                3           Children              8.0   
256371            TRACKS                3              Arson              7.0   
291193           RALSTON                3      Miscellaneous              9.0   
1286206        ADON ROAD                3          Lightning              1.0   
365024        2100 CENTE                3      Miscellaneous              9.0   
1276621             None                3     Debris Burning              5.0   
1063028             FOOT                3      Miscellaneous              9.0   
1234308    EARLY MORNING                3  Missing/Undefined             13.0   
1579610      BIRCH CREEK                4  Missing/Undefined             13.0   
1324066      JIMS BRANCH                4              Arson              7.0   
1227849            30305                4              Arson              7.0   
1210677         NE 30495                4              Arson              7.0   
1059663       INDIAN RUN                4      Miscellaneous              9.0   
356156        BUENAVISTA                4          Lightning              1.0   
305237        KILIOVILIK                4          Lightning              1.0   
1579611            SKULL                4          Lightning              1.0   
357487     ROAD 09-08 WF                5              Arson              7.0   
1286272          THUNDER                5      Miscellaneous              9.0   
1360116       COAL CREEK                5  Missing/Undefined             13.0   
1579684         HOPEWELL                5  Missing/Undefined             13.0   
1572579  SIXTEEN COMPLEX                7              Arson              7.0   

        STATE    COUNTY   LONGITUDE   LATITUDE  DISCOVERY_DATE DISCOVERY_TIME  \
182647     MN      None  -93.765300  46.212700       2449480.5           1450   
256371     SD      None -100.540700  45.670000       2452773.5           2120   
291193     OK      None  -96.752800  36.507800       2453385.5           1215   
1286206    WY  Campbell -105.276150  44.454340       2455430.5           2059   
365024     IN      None  -85.423200  38.937900       2452004.5           1300   
1276621    PA   FAYETTE  -79.827778  40.017222       2452352.5           1007   
1063028    CA        63 -121.052500  39.803889       2453557.5           1638   
1234308    AZ      None -109.923000  31.970500       2454581.5           0217   
1579610    OR   Malheur -117.290000  44.322200       2456090.5           1756   
1324066    WV   Wyoming  -81.349864  37.511007       2449661.5           1310   
1227849    OK     Adair  -94.743133  36.035600       2452723.5           1400   
1210677    OK      None  -94.898000  36.284700       2455501.5           0710   
1059663    CO       107 -107.408333  40.284167       2454741.5           1700   
356156     CA      None -119.522800  37.658900       2451763.5           1600   
305237     AK      None -157.519400  66.566100       2452071.5           2010   
1579611    OR   Malheur -117.838110  43.478810       2456162.5           1231   
357487     AL      None  -85.597800  34.449700       2453799.5           1145   
1286272    WY  Campbell -105.264870  43.740000       2455427.5           1345   
1360116    OK   Leflore  -94.866111  34.940000       2454486.5           1358   
1579684    PA      None  -75.761400  40.213300       2456026.5           1228   
1572579    CA    Colusa -122.344500  38.944440       2456174.5           0345   

         CONT_DATE CONT_TIME  DISCOVERY_DATETIME       CONT_DATETIME  \
182647   2449845.5      1600 1994-05-08 14:50:00 1995-05-08 16:00:00   
256371   2453139.5      2300 2003-05-14 21:20:00 2004-05-14 23:00:00   
291193   2453750.5      1330 2005-01-15 12:15:00 2006-01-15 13:30:00   
1286206  2455804.5      2059 2010-08-22 20:59:00 2011-08-31 20:59:00   
365024   2452379.5      1200 2001-04-05 13:00:00 2002-04-15 12:00:00   
1276621  2453448.

In [70]:
df[
    (df['DISCOVERY_DATETIME'].apply(lambda x: x.year) != df['CONT_DATETIME'].apply(lambda x: x.year))
    &
    (df['HOURS_TO_CONT'] > 200)
]

FIRE_NAME  FIRE_SIZE_CLASS STAT_CAUSE_DESCR  STAT_CAUSE_CODE  \
25280                DOMKE                7        Lightning              1.0   
34763             LICK RUN                2        Lightning              1.0   
48658                 None                1        Lightning              1.0   
80774             VICTORIA                1    Miscellaneous              9.0   
114369           FREIDLEIN                2         Campfire              4.0   
...                    ...              ...              ...              ...   
1782782               None                1    Miscellaneous              9.0   
1782784           WC - 131                3    Miscellaneous              9.0   
1782785     WATSON ORCHARD                1    Miscellaneous              9.0   
1782786  RISING EAGLE ROAD                5    Miscellaneous              9.0   
1782787        MONTE CARLO                1    Miscellaneous              9.0   

        STATE COUNTY   LONGITUDE   LATITUDE  DISCOVERY_DATE DISCOVERY_TIME  \
25280      WA      7 -120.599444  48.167222       2454317.5           1900   
34763      VA     15  -79.146944  38.368611       2454583.5           1630   
48658      CA   None -118.708333  36.295000       2448922.5           1800   
80774      CA   None -117.180000  34.243333       2449726.5           1409   
114369     AZ      5 -111.691667  35.291667       2451530.5           0950   
...       ...    ...         ...        ...             ...            ...   
1782782    NJ  Ocean  -74.275800  40.059000       2456982.5           1000   
1782784    WA    045 -123.472000  47.309700       2456977.5           1830   
1782785    WA    047 -119.897600  48.067900       2456854.5           1200   
1782786    WA    047 -120.146000  48.412900       2456870.5           1345   
1782787    NV   Elko -115.781791  40.850472       2456760.5           1548   

         CONT_DATE CONT_TIME  DISCOVERY_DATETIME       CONT_DATETIME  \
25280    2454472.5      0900 2007-08-05 19:00:00 2008-01-07 09:00:00   
34763    2454948.5      2100 2008-04-27 16:30:00 2009-04-27 21:00:00   
48658    2449287.5      1930 1992-10-27 18:00:00 1993-10-27 19:30:00   
80774    2450091.5      1500 1995-01-09 14:09:00 1996-01-09 15:00:00   
114369   2451547.5      1600 1999-12-18 09:50:00 2000-01-04 16:00:00   
...            ...       ...                 ...                 ...   
1782782  2457024.5      1000 2014-11-21 10:00:00 2015-01-02 10:00:00   
1782784  2457045.5      1430 2014-11-16 18:30:00 2015-01-23 14:30:00   
1782785  2457063.5      1200 2014-07-16 12:00:00 2015-02-10 12:00:00   
1782786  2457063.5      1200 2014-08-01 13:45:00 2015-02-10 12:00:00   
1782787  2457157.5      1328 2014-04-13 15:48:00 2015-05-15 13:28:00   

             TIME_TO_CONT  HOURS_TO_CONT  
25280   154 days 14:00:00           3710  
34763   365 days 04:30:00           8764  
48658   365 days 01:30:00           8762  
80774   365 days 00:51:00           8761  
114369   17 days 06:10:00            414  
...                   ...            ...  
1782782  42 days 00:00:00           1008  
1782784  67 days 20:00:00           1628  
1782785 209 days 00:00:00           5016  
1782786 192 days 22:15:00           4630  
1782787 396 days 21:40:00           9526  

[141 rows x 16 columns]

### Conclusion on outliers:

Probably best to drop ones with greater than 3000 hours. Looking at the fires that have > 1000 hours, most are within 2392. Safe to drop those with more as they are likely outliers.

In [69]:
#Save intermediate copy
df.to_pickle("data_with_target.pkl")

## Test model

In [72]:
from sklearn import tree
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn import metrics

In [157]:
# drop cols not used in model
drop_cols = ['FIRE_SIZE_CLASS', 'FIRE_NAME', 'STAT_CAUSE_DESCR', 'STATE', 'COUNTY', 'DISCOVERY_DATE',
            'DISCOVERY_DATETIME', 'CONT_DATE', 'CONT_DATETIME', 'TIME_TO_CONT']
X = df.drop(drop_cols, axis=1)
y = df['FIRE_SIZE_CLASS']

In [158]:
# create testing set
X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=True,test_size=0.2, random_state=1)

# create training and validation sets
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, shuffle=True, test_size=0.25, random_state=1)

In [159]:
md = 10
clf = DecisionTreeClassifier(max_depth=md).fit(X_train, y_train)

In [160]:
print("Tree depth: ", clf.get_depth())
print("# of leaves: ", clf.get_n_leaves())
print("Feature importance: ")
for feature, value in dict(zip(X.columns.values, clf.feature_importances_)).items():
    print(f"\t{feature:8}: {round(value,3)}")
print("Train score: ", clf.score(X_train, y_train))
print("Test score: ", clf.score(X_test, y_test))

Tree depth:  10
# of leaves:  953
Feature importance: 
	STAT_CAUSE_CODE: 0.078
	LONGITUDE: 0.39
	LATITUDE: 0.11
	DISCOVERY_TIME: 0.011
	CONT_TIME: 0.015
	HOURS_TO_CONT: 0.27
	STATE_CODE: 0.125
Train score:  0.6427897451994851
Test score:  0.6365791863319918


In [168]:
# remove "outliers" and see if improvement
X = df[df.HOURS_TO_CONT < 3000].drop(drop_cols, axis=1)
y = df[df.HOURS_TO_CONT < 3000]['FIRE_SIZE_CLASS']
# create testing set
X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=True,test_size=0.2, random_state=1)

# create training and validation sets
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, shuffle=True,test_size=0.25, random_state=1)

In [169]:
md = 10
clf = DecisionTreeClassifier(max_depth=md).fit(X_train, y_train)
print("Tree depth: ", clf.get_depth())
print("# of leaves: ", clf.get_n_leaves())
print("Feature importance: ")
for feature, value in dict(zip(X.columns.values, clf.feature_importances_)).items():
    print(f"\t{feature:8}: {round(value,3)}")
print("Train score: ", clf.score(X_train, y_train))
print("Test score: ", clf.score(X_test, y_test))

Tree depth:  10
# of leaves:  968
Feature importance: 
	STAT_CAUSE_CODE: 0.076
	LONGITUDE: 0.395
	LATITUDE: 0.112
	DISCOVERY_TIME: 0.01
	CONT_TIME: 0.017
	HOURS_TO_CONT: 0.27
	STATE_CODE: 0.121
Train score:  0.6435529121310739
Test score:  0.6375455193271276


## Try undersampling to improve performance

In [163]:
df.FIRE_SIZE_CLASS.value_counts()

2    386748
1    380835
3     92523
4     14622
5      8544
6      5584
7      3151
Name: FIRE_SIZE_CLASS, dtype: int64

In [180]:
# skewed data, let's undersample 1 and 2
sample2=df[df.FIRE_SIZE_CLASS == 2].sample(10000)
sample1=df[df.FIRE_SIZE_CLASS == 1].sample(10000)
sample3=df[df.FIRE_SIZE_CLASS == 3].sample(10000)

In [181]:
sampled_df= pd.concat([sample1, sample2, sample3,df[(df.FIRE_SIZE_CLASS != 1) & (df.FIRE_SIZE_CLASS != 2)]])
print(sampled_df.shape)
sampled_df.head()

(154424, 17)


FIRE_NAME  FIRE_SIZE_CLASS STAT_CAUSE_DESCR  STAT_CAUSE_CODE  \
312500   UNNAMED FIRE 0693                1            Arson              7.0   
1220677            JG PILE                1   Debris Burning              5.0   
352995               BLOCK                1         Campfire              4.0   
317188   UNNAMED FIRE 2503                1          Smoking              3.0   
70804                 None                1        Lightning              1.0   

        STATE     COUNTY   LONGITUDE   LATITUDE  DISCOVERY_DATE  \
312500     CA       Kern -119.406903  35.054822       2453755.5   
1220677    WI       Dane  -89.764020  43.049200       2454936.5   
352995     TX       None -101.664400  35.721100       2455102.5   
317188     CA  Riverside -117.275714  33.749322       2454262.5   
70804      CA       None -122.553333  41.048333       2449623.5   

        DISCOVERY_TIME  CONT_DATE CONT_TIME  DISCOVERY_DATETIME  \
312500            1400  2453755.5      1400 2006-01-20 14:00:00   
1220677           1310  2454936.5      1310 2009-04-15 13:10:00   
352995            1030  2455102.5      1040 2009-09-28 10:30:00   
317188            0924  2454262.5      0925 2007-06-11 09:24:00   
70804             1730  2449623.5      1800 1994-09-28 17:30:00   

              CONT_DATETIME TIME_TO_CONT  HOURS_TO_CONT  STATE_CODE  
312500  2006-01-20 14:00:00     00:00:00              0           0  
1220677 2009-04-15 13:10:00     00:00:00              0          34  
352995  2009-09-28 10:40:00     00:10:00              0          15  
317188  2007-06-11 09:25:00     00:01:00              0           0  
70804   1994-09-28 18:00:00     00:30:00              0           0

In [182]:
# remove "outliers" and see if improvement
X = sampled_df[df.HOURS_TO_CONT < 3000].drop(drop_cols, axis=1)
y = sampled_df[df.HOURS_TO_CONT < 3000]['FIRE_SIZE_CLASS']
# create testing set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, shuffle=True, random_state=1)

# create training and validation sets
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, test_size=0.25, shuffle=True,random_state=1)

C:\Users\somet\anaconda3\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  
C:\Users\somet\anaconda3\lib\site-packages\ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


In [186]:
md = 5
clf = DecisionTreeClassifier(max_depth=md).fit(X_train, y_train)
print("Tree depth: ", clf.get_depth())
print("# of leaves: ", clf.get_n_leaves())
print("Feature importance: ")
for feature, value in dict(zip(X.columns.values, clf.feature_importances_)).items():
    print(f"\t{feature:8}: {round(value,3)}")
print("Train score: ", clf.score(X_train, y_train))
print("Test score: ", clf.score(X_test, y_test))

Tree depth:  5
# of leaves:  32
Feature importance: 
	STAT_CAUSE_CODE: 0.05
	LONGITUDE: 0.35
	LATITUDE: 0.008
	DISCOVERY_TIME: 0.0
	CONT_TIME: 0.017
	HOURS_TO_CONT: 0.411
	STATE_CODE: 0.164
Train score:  0.6781333419792713
Test score:  0.6742210550205882


**Conclusion**: performance improved a lot

## Even sampling across everything

Sample so everything is max the lowest count of fire size classes.

In [172]:
# skewed data, let's undersample 1 and 2
sample1=df[df.FIRE_SIZE_CLASS == 1].sample(3151)
sample2=df[df.FIRE_SIZE_CLASS == 2].sample(3151)
sample3=df[df.FIRE_SIZE_CLASS == 3].sample(3151)
sample4=df[df.FIRE_SIZE_CLASS == 4].sample(3151)
sample5=df[df.FIRE_SIZE_CLASS == 5].sample(3151)
sample6=df[df.FIRE_SIZE_CLASS == 6].sample(3151)
sample7=df[df.FIRE_SIZE_CLASS==7]

sampled_df= pd.concat([sample1, sample2, sample3, sample4, sample5, sample6, sample7])
print(sampled_df.shape)

(22057, 17)


In [177]:
# remove "outliers" and see if improvement
X = sampled_df[df.HOURS_TO_CONT < 8000].drop(drop_cols, axis=1)
y = sampled_df[df.HOURS_TO_CONT < 8000]['FIRE_SIZE_CLASS']
# create testing set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, shuffle=True, random_state=1)

# create training and validation sets
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, test_size=0.25, shuffle=True,random_state=1)

C:\Users\somet\anaconda3\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  
C:\Users\somet\anaconda3\lib\site-packages\ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


In [178]:
md = 10
clf = DecisionTreeClassifier(max_depth=md).fit(X_train, y_train)
print("Tree depth: ", clf.get_depth())
print("# of leaves: ", clf.get_n_leaves())
print("Feature importance: ")
for feature, value in dict(zip(X.columns.values, clf.feature_importances_)).items():
    print(f"\t{feature:8}: {round(value,3)}")
print("Train score: ", clf.score(X_train, y_train))
print("Test score: ", clf.score(X_test, y_test))

Tree depth:  10
# of leaves:  717
Feature importance: 
	STAT_CAUSE_CODE: 0.046
	LONGITUDE: 0.216
	LATITUDE: 0.141
	DISCOVERY_TIME: 0.06
	CONT_TIME: 0.064
	HOURS_TO_CONT: 0.427
	STATE_CODE: 0.046
Train score:  0.5544554455445545
Test score:  0.4180457946043981


**Conclusion**: performance degraded drastically by overrepresenting the rarer fire classes

## What next

Select best max depth decision tree?

Ensemble method Random Forest?